js/style/style.js

```diff
@@ -370,7 +370,10 @@ Style.prototype = util.inherit(Evented, {
         layer = this.getReferentLayer(layer);
         layer.setLayoutProperty(name, value);
         this._broadcastLayers();
-        this.sources[layer.source].reload();
+        if (layer.source) {
```

Comment:
Oh wait, `setPaintProperty` won't have this issue. `setFilter` does, but background layers can't have filters.